slang-api/src/main/java/com/sonarsource/slang/api/ClassDeclarationTree.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.api;
+
+public interface ClassDeclarationTree extends Tree {
+
+  Tree classTree();
```

Comment:
I would have liked to have the `IdentifierTree` in the `ClassDeclarationTree ` too. Otherwise there is no way to retrieve the name of the class easily. WDYT ?

Reply:
For now, we don't need the name of the class as we only need classes for metrics; adding names will add extra complexity to the Kotlin converter which I would like to avoid for now.

Reply:
Ok. We'll need it at some point for the rule about class name, but we can update API at that time